denylist/src/denylist.rs

```diff
@@ -122,21 +123,21 @@ pub fn filter_from_bin(bin: &Vec<u8>) -> Result<Xor32> {
     let _version = buf.get_u8();
     let signature_len = buf.get_u16_le() as usize;
     let signature = buf.copy_to_bytes(signature_len).to_vec();
-    let pubkey = PublicKey::from_str(PUB_KEY_B58)?;
-    match pubkey.verify(buf, &signature) {
-        Ok(_) => {
-            tracing::info!("updating filter to latest");
-            let _serial = buf.get_u32_le();
-            let xor = bincode::deserialize::<Xor32>(buf)?;
-            Ok(xor)
-        }
-        Err(_) => {
-            tracing::warn!("filter signature verification failed");
-            Err(Error::InvalidBinary(
-                "filter signature verification failed".to_string(),
-            ))
+    for pubkey in valid_sign_keys {
+        match pubkey.verify(buf, &signature) {
+            Ok(_) => {
+                tracing::info!("updating filter to latest");
+                let _serial = buf.get_u32_le();
+                let xor = bincode::deserialize::<Xor32>(buf)?;
```

Comment:
Since buf is just a slice, you can create a subslice and I think that's better than manually moving the pointer forward:
```rust
buf = buf[4..];
```

Reply:
you mean instead of the `buf.get_u32_le()`?

Reply:
huh? you _want_ to move the cursor forward?

Reply:
> huh? you _want_ to move the cursor forward? which get_u32_le would do no?

I mean manually as in "through a contrived function that doesn't explain what it's doing"

Reply:
it's documented in get_u32_le and part of Buf behavior? But if we wanted to make that absolutely clear here and since we're not using the result, you could just call `advance(4)` instead?

Reply:
I like that idea the best, so I made the change 923a641